pytorch_toolkit/formula_recognition/im2latex/models/im2latex_model.py

```diff
@@ -48,6 +48,13 @@ def forward(self, hidden, context, output, row_enc_out, tgt):
 
     def __init__(self, backbone_type, backbone, out_size, head):
         super().__init__()
+        bb_out_channels = backbone.get("output_channels", 512)
+        head_in_channels = head.get("encoder_input_size", 512)
+        assert bb_out_channels == head_in_channels, f"""
+        Number of output channels in the backbone ({bb_out_channels}) must be equal
+        to the number of input channels in the head ({head_in_channels}) in case last conv
+        is disabled
+        """
```

Comment:
For this to be a valid check, you should guarantee that `backbone.output_channels` always stores the actual number of the channels in the output tensor. Please make sure that this is what is happening there.

Reply:
Added check in the `resnet.py` that actual number of output channels is equal to the number from the config file